main.go

```diff
@@ -242,30 +312,25 @@ func deploy(clearedFilesToDeploy []string, config Config) (map[string]string, er
 				return nil, fmt.Errorf("deploy failed, error: %s", err)
 			}
 
-			if installPage != "" {
-				publicInstallPages[filepath.Base(pth)] = installPage
-			}
-		case ".apk":
-			log.Donef("Uploading apk file: %s", pth)
-
-			installPage, err := uploaders.DeployAPK(pth, config.BuildURL, config.APIToken, config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled)
-			if err != nil {
-				return nil, fmt.Errorf("deploy failed, error: %s", err)
-			}
-
 			if installPage != "" {
 				publicInstallPages[filepath.Base(pth)] = installPage
 			}
 		case ".aab":
 			log.Donef("Uploading aab file: %s", pth)
 
-			installPage, err := uploaders.DeployAAB(pth, config.BuildURL, config.APIToken, config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled)
-			if err != nil {
+			if err := uploaders.DeployAAB(pth, androidArtifacts, config.BuildURL, config.APIToken,
+				config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled); err != nil {
 				return nil, fmt.Errorf("deploy failed, error: %s", err)
 			}
 
-			if installPage != "" {
-				publicInstallPages[filepath.Base(pth)] = installPage
+			universalAPKPair, ok := aabAPKPairs[pth]
+			if !ok {
+				return nil, fmt.Errorf("no iniversal apk pair found for aab: %s", pth)
+			}
+
+			universalAPKInstallPage, ok := apkInstallPages[filepath.Base(universalAPKPair)]
```

Comment:
this can be moved to the if and also don;t forget to remove `ok` or use it for validation